waku/waku_rln_relay/rln_relay.nim

```diff
@@ -392,6 +393,33 @@ proc generateRlnValidator*(
 
   return validator
 
+proc monitorEpochs(wakuRlnRelay: WakuRLNRelay): Future[void] {.async.} =
+  var nextEpochTime = epochTime()
+  var lastEpoch = wakuRlnRelay.calcEpoch(epochTime())
+  var firstChangeDetected = false
+
+  while true:
+    try:
+      let currentEpoch = wakuRlnRelay.calcEpoch(epochTime())
+
+      if currentEpoch != lastEpoch:
+        nextEpochTime = nextEpochTime + float64(wakuRlnRelay.rlnEpochSizeSec)
```

Comment:
Isn't there a way to calculate the epoch time in which the RLN epoch started? Now the time won't drift but we can be off up to 500 ms forever.

If we use as reference the time in which the RLN epoch started instead of the time we are checking, it will make it much more accurate :)

Reply:
Ok there isn't a simple function to get the next epoch, but it is also not too hard to do

https://github.com/waku-org/nwaku/blob/8f2bd39f2304628ad27b42c3964b4a614292b56a/waku/waku_rln_relay/rln_relay.nim#L93-L97

Here's how RLN epochs are calculated as a function of time.

To find the time where the next epoch starts, we need the minimum `t` greater than the current time such that

```
uint64(t / rlnPeer.rlnEpochSizeSec.float64)
```

results in a bigger epoch than the current one.

So basically, we need the smallest multiple of `rlnPeer.rlnEpochSizeSec` that is bigger than the current time.

Even the brute force way, which is adding 1 in a loop to `t` until `t % rlnPeer.rlnEpochSizeSec.float64 == 0` does the job hahaha

Anyway, don't want to overcomplicate things, so I leave it to your criteria :)